include/picongpu/CMakeLists.txt

```diff
@@ -89,7 +89,7 @@ function(PIC_dependency_set_status name)
     else()
         set(value FALSE)
     endif()
-    if(${value} AND NOT PIC_HAVE_${name}_LOCKED)
+    if(${value} AND NOT PIC_HAVE_${name}_LOCKED AND PIC_SEARCH_${name})
```

Comment:
Is this evaluated from left to right or right o left because the logic results differ for 
T, T, F etc.
 - right to left = `(T and not (T and F)) = (T and not F) = (T and T) = T`
 - left to right = ` `((T and not T) and F) = (T and F) and F = F and F = F`
 Could adding parentheses solve this required knowledge about CMake execution order?

Reply:
it is from left to right. But the result is equal from both sides because `NOT` is only bound to `PIC_HAVE_${name}_LOCKED`

Reply:
precedence of CMake operators in if conditions: https://cmake.org/cmake/help/latest/command/if.html#condition-syntax

Reply:
Since `NOT` operator behavior is well defined, we would forego setting additional (unnecessary) parenthesis. We also don't set additional parenthesis in other places.